### PR TITLE
Implemented fixes for errors found by the W3C css validator. 

### PR DESCRIPTION
**Pull request by wagnered**
With first try ccommit wss orphaned.

### CODE REVIEW

**Comment by lachlan-00** [comment]
just updated the markdown for the api from the changes i'm putting in

**Review by usox — APPROVED** [review]

**Comment by wagnered** [comment]
@usox I didn't realize I needed your approval.😄

I worked for Digital Equipment corp for about 10 years before deciding  that I had enough of half-dozen others having authority over me. So I joined the maritime service as a radio operator/electronics officer, in which there was only one person over me: the captain,  and for almost twenty years he advised me only when necessary.

If I am wrong about this, I apologize,  I realize that you probably work in an environment that, either by doctrine, or by culture, consensus might be de rigueur.


And between DEC and the Maritime service, I procured a Master in counseling psychology.

**Comment by usox** [comment]
> @usox I didn't realize I needed your approval.smile

Oh my. No, you surely don't need my approval.
I'm just using githubs review workflow/functionality in pull requests which result in an automatic `approval` after a code "review". So it's more like a signal that says "Hey there, I just took a look at your changes and found no nasty bugs in it. You're good to go" - maybe the wording isn't ideal but this is a github thingy.
If I'm working in/for a software project, I'm used to take a quick look over PR's to help finding bugs or just to give some hints to improve the code. But this doesn't (or should not) manifest any kind of hierarchy - this is just a few people/a team talking over code.
If you don't want me to have a look over your or anyones code, please tell me and I will stop doing it. I'm ok with it.

> If I am wrong about this, I apologize, I realize that you probably work in an environment that, either by doctrine, or by culture, consensus might be de rigueur.

In the software companies and opensource teams where I have worked in the last 20 years, such code reviews have been daily business. The whole team is responsible for reviewing every change to ensure quality and maintainability. But as said, this is not about hierarchy and just about quality and learning.
So, for me this is just an automatism to watch out for PRs and have a quick look...

> And between DEC and the Maritime service, I procured a Master in counseling psychology.

I was always interested in electronics (building circuits and such), but I never had enough time to dive deep into the topic.

I have the feeling that somehow we both got off to an unlucky start (which is certainly also due to my sub-optimal communication skills). But I hope that we will get along with each other in the future.

**Comment by wagnered** [comment]
There should be no problem.  I'm highly reactive to authority figure, but I can keep it in check.

I'm also a licensed amateur radio operater and was in bremerhaven

**Comment by lachlan-00** [comment]
is this accidentally closed?

**Comment by wagnered** [comment]
Yes, thanks for bringing it to my attention
.

**Comment by lachlan-00** [comment]
thanks, i haven't had any real time this week while at work beyond a few little checks but i want to do these on the weekend

File: themes/reborn/templates/default.css [hunk 1]
```diff
@@ -1007,117 +1018,64 @@ table.tabledata .th-bottom:focus {
...
 }
 
-table.tabledata .cel_agent img:hover {
-    cursor: help;
+table.gatherart .th-top:hover,
+table.gatherart .th-bottom:hover,
+table.gatherart .th-top:focus,
+table.gatherart .th-bottom:focus {
+    background-image: none;
+    background-repeat: no-repeat;
```
Comment (AshotN) [inline themes/reborn/templates/default.css:1074]: If there is no background image. What is the repeat for

File: themes/reborn/templates/light.css [hunk 2]
```diff
@@ -404,6 +404,10 @@ table.tabledata tbody .cel_cover img:hover {
     border-color: #1990db;
 }
 
+input:invalid {
+   border: 4px solid red;;
```
Comment (AshotN) [inline themes/reborn/templates/light.css:408]: Extra ;

File: themes/reborn/templates/default.css [hunk 3]
```diff
@@ -1007,117 +1018,64 @@ table.tabledata .th-bottom:focus {
...
-
-table.tabledata tbody .cel_play_content {
-    display: block;
-}
-
-table.tabledata tbody .cel_play_hover {
-    display: none;
+table.gatherart {
+    width: 100%;
+    /* table-layout: fixed; */
```
Comment (AshotN) [inline themes/reborn/templates/default.css:1023]: Unneeded comment line.

**Review by AshotN — COMMENTED** [review]
I marked some things I thought odd.

**Comment by wagnered** [comment]
Hello @AshotN,

It's more about human memory/attention span.

At least the double semicolon was for me.

The other oddities you refer to were probably from inexperience. 

 Since the last read overwrites the previous, in both css and php, and the interface is as expected, I wasn't about to make any changes: my only concern was for explicit errors.

But

```
 	input:invalid {
		   border: 4px solid red;;
        }
```
Should not have been in the develop branch as it was meant for a different PR.  If you think I should remove it , I'd be happy to.

You are more familiar with css than i: you are welcome  to clean up the three css theme files.

**Comment by lachlan-00** [comment]
okay, putting the css on my server now. i'll take the no background part out that ashot has mentioned

**Comment by lachlan-00** [comment]
not sure what part but the hover is probably required for the play buttons. with these files it doesn't hide the play buttons and they're always visible
image

**Comment by kuzi-moto** [comment]
I think there is some great work being put into this, but as I have looked into the CSS/HTML pretty extensively and feel like it really should just be re-written.

I think it's a similar argument as with moving to Laravel, but I think different enough as in this case it's just cosmetic, and needed a face-lift anyway.

Ampache CSS relies heavily on high specifity, such as using IDs, using many selectors, or worst yet `!important`. This is generally against CSS best practices. If we're working on modernizing Ampache seems like a good place to start, especially since we could create a responsive design, and even someday make it into a [Progressive Web Application](https://en.m.wikipedia.org/wiki/Progressive_web_application) allowing Ampache to have a native app without even needing to create separate platform specific apps.

Of course any work on Ampache is great and re-writing will take time so makes sense if that's not the path taken now. Just wanted to throw those ideas out there!

**Comment by wagnered** [comment]
I would like to revert default.css. I would like to remove all references to "gatherart" which was meant for a different feature.  I had just copied the table data class info and made a few changes.  Not sure how the changes got into the develop branch.

**Comment by AshotN** [comment]
> I think there is some great work being put into this, but as I have looked into the CSS/HTML pretty extensively and feel like it really should just be re-written.
> 
> I think it's a similar argument as with moving to Laravel, but I think different enough as in this case it's just cosmetic, and needed a face-lift anyway.
> 
> Ampache CSS relies heavily on high specifity, such as using IDs, using many selectors, or worst yet `!important`. This is generally against CSS best practices. If we're working on modernizing Ampache seems like a good place to start, especially since we could create a responsive design, and even someday make it into a [Progressive Web Application](https://en.m.wikipedia.org/wiki/Progressive_web_application) allowing Ampache to have a native app without even needing to create separate platform specific apps.
> 
> Of course any work on Ampache is great and re-writing will take time so makes sense if that's not the path taken now. Just wanted to throw those ideas out there!

The React Interface im working on is a rewrite. It is also a PWA and I am transitioning it to CSS modules. The CSS Is also written in Stylus. 

You are welcome to help me with the design or anything else regarding it.

I don't think its worth the effort to fix Reborn. Way to much work for such an old interface, Reborn doesn't work for shit on mobile. But it also has soooo much functionality that it a new interface will be a slow roll out that will over time aim to replace Reborn.

**Comment by lachlan-00** [comment]
will test this one again in a few hrs with spotify

**Comment by wagnered** [comment]
With Spotify?  Are you sure?  This is a PR into develop.  I removed all refs to gather art. In this PR.  I'll check again to make sure I made the correct decision on that.

I'll soon have the show_get_art template laid out with complete validation for the gather_spotify improvement PR.

I noticed that `gather_musicbrainz` rejects searches that don't include the mbid, so I'll implement validation for that.

**Comment by lachlan-00** [comment]
(with the spotify change as well.)

**Pull request merged**